Fix build on Yosemite and later by declaring readlinkat the same way as Yosemite does
https://github.com/gittup/tup/commit/9812548b4d3833644f0cf8f70f62628a75ffbb56
--- src/compat/macosx.h.orig
+++ src/compat/macosx.h
@@ -24,6 +24,7 @@
 #include <stdlib.h>
 #include <sys/stat.h>
 #include <dirent.h>
+#include <unistd.h>
 
 /* MacOSX 10.6 does not have *at() functions */
 DIR *fdopendir(int fd);
@@ -33,7 +34,7 @@ int fchownat(int dirfd, const char *pathname, uid_t owner, gid_t group, int flag
 int fstatat(int dirfd, const char *pathname, struct stat *buf, int flags);
 int mkdirat(int dirfd, const char *pathname, mode_t mode);
 int openat(int dirfd, const char *pathname, int flags, ...);
-int readlinkat(int dirfd, const char *pathname, char *buf, size_t bufsiz);
+ssize_t readlinkat(int dirfd, const char *pathname, char *buf, size_t bufsiz);
 int renameat(int olddirfd, const char *oldpath, int newdirfd, const char *newpath);
 int symlinkat(const char *oldpath, int newdirfd, const char *newpath);
 int unlinkat(int dirfd, const char *pathname, int flags);
--- src/compat/readlinkat.c.orig
+++ src/compat/readlinkat.c
@@ -23,7 +23,7 @@
 #include <unistd.h>
 #include "dir_mutex.h"
 
-int readlinkat(int dirfd, const char *pathname, char *buf, size_t bufsiz)
+ssize_t readlinkat(int dirfd, const char *pathname, char *buf, size_t bufsiz)
 {
 	int rc;
 
